Eagerly compute witness when sequencing (fork12)

cffls opened the pull request:
Before this change, the witness of a batch is computed only right before being sent to an executor. This mechanism can result in a longer time in chain unwinding if the witness to be computed is far away from the head. For example, if the batch to be verified is 1000 batches behind the tip of the sequencer, the sequencer will need to unwind the same amount batches before computing the witness. Overall, the unnecessary computation complexity of this behavior is `O(n^2)`, where `n` is the number of pending batches to be verified.
We can remove this redundant computation by always eagerly computing the witness when a batch is closed, without delaying to the verification promise and therefore avoid chain unwinding overall.

hexoscott commented on zk/legacy_executor_verifier/legacy_executor_verifier.go:
```diff
@@ -262,12 +265,15 @@ func (v *LegacyExecutorVerifier) VerifyAsync(request *VerifierRequest) *Promise[
-		if err != nil {
-			return verifierBundle, err
-		}
+		if request.witness == nil {
```
Could the PR be made simpler by just moving the witness generation to before the line 'e.AcquireAccess'?  This would have the same effect I think.

cffls replied:
That's a great idea. I've tested your suggestion and it works smoothly. Updated the PR.

cffls committed:
Eagerly compute witness when sequencing

Before this change, the witness of a batch is computed only right before being sent to an executor. This mechanism can result in a longer time in chain unwinding if the witness to be computed is far away from the head. For example, if the batch to be verified is 1000 batches behind the tip of the sequencer, the sequencer will need to unwind the same amount batches before computing the witness. Overall, the unnecessary computation complexity of this behavior is O(n^2), where n is the number of pending batches to be verified.
We can remove this redundant computation by always eagerly computing the witness when a batch is closed, without delaying to the verification promise and therefore avoid chain unwinding overall.
